core/templates/pages/story-editor-page/editor-tab/story-editor.directive.ts

```diff
@@ -135,6 +138,7 @@ angular.module('oppia').directive('storyEditor', [
             });
 
             modalInstance.result.then(function(title) {
+              StoryEditorStateService.setExpIdsChanged();
```

Comment:
Why we are calling this before line 142? (Also, I was expecting `deleteStoryNode` to make this call internally)

Reply:
Done, moved these to inside StoryUpdateService.